.github/workflows/pull-request.yml

```diff
@@ -0,0 +1,52 @@
+name: PR Check
+
+on: [pull_request]
+
+# When a new revision is pushed to a PR, cancel all in-progress CI runs for that
+# PR. See https://docs.github.com/en/actions/using-jobs/using-concurrency
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  go-lint:
+    name: Go Lint
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+
+      - name: Golang Style and Lint Check
+        run: docker run --rm -v $(pwd):/app -w /app golangci/golangci-lint:v1.46.2 golangci-lint run -v
```

Comment:
So I wonder why you are choosing to run the docker image, rather than the golangci-lint GitHub action from the marketplace? Does the docker version also annotate the PR with inline comments the way the action marketplace golangci-lint does?

Reply:
I'll admit, I just copied that part of the workflow over wholesale from [OPA's `pull-request` workflow](https://github.com/philipaconrad/opa/blob/main/.github/workflows/pull-request.yaml), and lifted up the underlying shell command for the `make check` target (the Docker command you saw).  :sweat_smile: 

I'll try out the Github action, and see if that works equally well. The less fiddly commands in those workflows, the less stuff that can break.

Reply:
Looks like the Github actions version works okay. I pinned it to `latest` on the linter version.